src/main/kotlin/interactor/Consumption.kt

```diff
@@ -5,58 +5,39 @@ import model.CityEntity
             .getAllCitiesData()
-            .filter(::excludeNullValue)
-            .sortedBy{it.realEstatesPrices.apartment3BedroomsOutsideOfCentre}
-        cities.forEach {
-             val servicesCoast= servicesCoast(it)
-             val remainder= remainder(it,servicesCoast)
-             if(isSuitable(servicesCoast,remainder)){
-                 if (x.contains(remainder)) x[remainder]?.add(it.cityName)
-                 else x[remainder] = mutableListOf(it.cityName)
+            .filter(::excludeNullOrZeroValue)
+            .forEach {
+             val remainder= remainder(it)
+             if(remainder>(250).toFloat()){
+                 if (x.contains(remainder)) x[remainder]?.add(it)
+                 else x[remainder] = mutableListOf(it)
              }
          }
         return x.toSortedMap()
             .values
-            .last()
-            .toString()
+            .last()[0]
     }
 }
-private fun excludeNullValue(city: CityEntity): Boolean {
-    return city.averageMonthlyNetSalaryAfterTax != null
-            && city.dataQuality
-            &&city.realEstatesPrices.apartment3BedroomsOutsideOfCentre != null
-            &&city.foodPrices.riceWhite1kg != null
-            &&city.foodPrices.chickenFillets1kg != null
-            &&city.foodPrices.beefRound1kgOrEquivalentBackLegRedMeat != null
-            &&city.foodPrices.localCheese1kg != null
-            &&city.foodPrices.loafOfFreshWhiteBread500g != null
-            &&city.servicesPrices.basicElectricityHeatingCoolingWaterGarbageFor85m2Apartment != null
-            &&city.servicesPrices.internet60MbpsOrMoreUnlimitedDataCableAdsl != null
-            &&city.servicesPrices.internationalPrimarySchoolYearlyForOneChild != null
-            &&city.servicesPrices.oneMinOfPrepaidMobileTariffLocalNoDiscountsOrPlans != null
+private fun excludeNullOrZeroValue(city: CityEntity): Boolean {
+    return !(city.averageMonthlyNetSalaryAfterTax == null
+            || city.realEstatesPrices.apartment3BedroomsOutsideOfCentre == null
+            || city.foodPrices.riceWhite1kg == null
+            || city.foodPrices.chickenFillets1kg == null
+            || city.foodPrices.beefRound1kgOrEquivalentBackLegRedMeat == null
+            || city.foodPrices.localCheese1kg == null
+            || city.foodPrices.loafOfFreshWhiteBread500g == null)
 }
-
-private fun servicesCoast(city:CityEntity):Float{
-    return (city.servicesPrices.basicElectricityHeatingCoolingWaterGarbageFor85m2Apartment!!
-        +(city.servicesPrices.internet60MbpsOrMoreUnlimitedDataCableAdsl!!)
-        +(city.servicesPrices.internationalPrimarySchoolYearlyForOneChild!!/6)
-        +(city.servicesPrices.oneMinOfPrepaidMobileTariffLocalNoDiscountsOrPlans!!))
-    }
-private fun remainder(city:CityEntity,servicesCoast:Float):Float{
+private fun remainder(city:CityEntity):Float{
 
     return( city.averageMonthlyNetSalaryAfterTax!!*2
             -city.realEstatesPrices.apartment3BedroomsOutsideOfCentre!!
```

Comment:
line 37 you have apartments that inside the city and outside the city you should get the minimum amount between them it is just a logic doesn't mean something to me just to make you aware only..